### ISSUE TITLE
Missing data points not shown correctly in 2D plot

### ISSUE THREAD

**Issue by mcrot** [ContactEngineering/TopoBank#699]
I've made a txt file "missing.xyz" with missing data:
```
0 0 1                                                                                                    
0 1 1                                                                                                    
0 2 1                                                                                                    
0 3 1                                                                                                                                                                                                              
1 0 NaN                                       
1 1 1                                                                                                    
1 2 1                                                                                                    
1 3 1                                               
2 0 1                                                                                                    
2 1 NaN                                             
2 2 2                                     
2 3 2                                               
3 0 2             
3 1 2        
3 2 2                                               
3 3 2
```
When loading this with `SurfaceTopography` , the NaNs are correctly represented as masked array:

```python
In [1]: from SurfaceTopography.IO import open_topography
                                                    
In [2]: r = open_topography("missing.xyz")          
                                                    
In [3]: r.format()
Out[3]: 'xyz'                                       
                                                    
In [4]: t = r.topography()                                                                                                                                                                                         
In [5]: t.heights()
Out[5]: 
masked_array(
  data=[[1.0, 1.0, 1.0, 1.0],
        [--, 1.0, 1.0, 1.0],
        [1.0, --, 2.0, 2.0],
        [2.0, 2.0, 2.0, 2.0]],
  mask=[[False, False, False, False],
        [ True, False, False, False],
        [False,  True, False, False],
        [False, False, False, False]],
  fill_value=1e+20)                              
```

The fill value is "1e+20". When saving as NetCDF
```python
In [6]: t.to_netcdf('missing.nc')

In [7]: t2 = open_topography('missing.nc').topography()

In [8]: t2.heights()
Out[8]: 
masked_array(
  data=[[1.0, 1.0, 1.0, 1.0],
        [--, 1.0, 1.0, 1.0],
        [1.0, --, 2.0, 2.0],
        [2.0, 2.0, 2.0, 2.0]],
  mask=[[False, False, False, False],
        [ True, False, False, False],
        [False,  True, False, False],
        [False, False, False, False]],
  fill_value=1e+20)
```
the masked arrays and fill values stay the same. So NetCDF does not change anything.

However, when plotted, the missing values are represented as 1:
![grafik](https://user-images.githubusercontent.com/9113541/126955215-6a4d770d-0310-467a-b748-a611385552ab.png)

Not sure whether this is wrong because of the "fill_value" of the masked array or if there is something with the plot.
If the plot had used the fill value, the maximum value would be much higher in the plot.

What is expected here, @pastewka , @AntoineSIMTEK ? A black spot?

**Comment by AntoineSIMTEK** [comment]
I guess 1e+20  is NetCDF's representation of NaN. 

Concerning the plot, I would expect white points where the area is masked. 

I think matplotlib is able to do that. Not sure about Bokeh.

**Comment by mcrot** [comment]
Hm .. I'm no longer sure that it's the plotting. It happens somehow with the caching and squeezed files
that the mask is lost. I'm still figuring out what's going on here.

**Comment by AntoineSIMTEK** [comment]
Ah ok. I remember that masked arrays caused me some big trouble in the past.
They didn't work as expected. I think it was with the interaction potentials

**Comment by AntoineSIMTEK** [comment]
I think it was that issue: https://github.com/pastewka/PyCo/issues/98

**Comment by mcrot** [comment]
I've found it: When saving a **detrended** topography to netCDF, the masked array is lost and replaced by strange values (here "1"). This is what I do in Topobank when saving squeezed datafiles.

**Comment by mcrot** [comment]
See issue here: https://github.com/ContactEngineering/SurfaceTopography/issues/126

Actually this only happens, if the topography is **scaled and detrended**.

**Comment by mcrot** [comment]
Bokeh seems to be capable of plotting masked arrays:
![grafik](https://user-images.githubusercontent.com/9113541/126970288-e689b191-09ff-4e35-ad6a-d352f9b0e351.png)

**Comment by mcrot** [comment]
When there is a new version of `SurfaceTopography` we need to find out all measurements with missing data points
and renew all squeezed files and all analyses for them. Probably it's worth to extend the existing management command(s),
because sth could happen again.

**Comment by mcrot** [comment]
At least with SurfaceTopography 0.95.2 this is fixed. Will be released with TopoBank 0.16.0. Closing.
